ArduPlane/landing.cpp

```diff
@@ -25,6 +25,26 @@ bool Plane::verify_land()
         return true;
     }
 
+    // when aborting a landing, mimic the verify_takeoff with steering hold. Once
+    // the altitude has been reached, restart the landing sequence
+    if (flight_stage == AP_SpdHgtControl::FLIGHT_LAND_ABORT) {
+
+        throttle_suppressed = false;
+        auto_state.land_complete = false;
+        nav_controller->update_heading_hold(steer_state.hold_course_cd);
+
+        // see if we have reached abort altitude
+        if (adjusted_relative_altitude_cm() > auto_state.takeoff_altitude_rel_cm) {
+            steer_state.hold_course_cd = -1;
+            next_WP_loc = current_loc;
+            mission.stop();
+            restart_landing_sequence();
```

Comment:
The only way this would fail is if you've never flown to another waypoint with a lat/lng. For example, if you did TAKEOFF then LAND then it would not abort but a mission like that doesn't make sense. However, I agree that having a fly-away is horrible and should never be possible. Sure, I can see a fall-back being RTL. Wouldn't hurt. I'll put one in.